corehq/apps/app_manager/suite_xml/post_process/resources.py

```diff
@@ -0,0 +1,65 @@
+from collections import Counter
+from django.db import models
+
+from corehq.apps.app_manager.exceptions import ResourceOverrideError
+from corehq.apps.app_manager.suite_xml.contributors import PostProcessor
+from corehq.apps.app_manager.suite_xml.sections.resources import FormResourceContributor
+from corehq.apps.app_manager.suite_xml.xml_models import XFormResource
+
+
+class ResourceOverride(models.Model):
+    domain = models.CharField(max_length=255, null=False)
+    app_id = models.CharField(max_length=255, null=False)
+    # Type of resource, e.g., xform. Populated by the root_name of the relevant suite_xml.xml_models class.
+    root_name = models.CharField(max_length=32, null=False)
+    pre_id = models.CharField(max_length=255, null=False)
+    post_id = models.CharField(max_length=255, null=False)
+
+    class Meta(object):
+        unique_together = ('domain', 'app_id', 'root_name', 'pre_id')
+
+
+def add_xform_resource_overrides(domain, app_id, pre_to_post_map):
+    overrides_by_pre_id = get_xform_resource_overrides(domain, app_id)
+    for pre, post in pre_to_post_map.items():
+        if pre in overrides_by_pre_id:
+            if post != overrides_by_pre_id[pre].post_id:
+                raise ResourceOverrideError("Cannot change override of {}".format(pre))
+        else:
+            override = ResourceOverride.objects.create(
+                domain=domain,
+                app_id=app_id,
+                root_name=XFormResource.ROOT_NAME,
+                pre_id=pre,
+                post_id=post,
+            )
+            override.save()
+
+
+def get_xform_resource_overrides(domain, app_id):
+    return {
+        override.pre_id: override
+        for override in ResourceOverride.objects.filter(
+            domain=domain,
+            app_id=app_id,
+            root_name=XFormResource.ROOT_NAME,
+        )
+    }
+
+
+class ResourceOverrideHelper(PostProcessor):
+
+    def update_suite(self):
+        """
+        Applies manual overrides of resource ids.
+        """
+        overrides_by_pre_id = get_xform_resource_overrides(self.app.domain, self.app.master_id)
+        resources = getattr(self.suite, FormResourceContributor.section_name)
+        for resource in resources:
+            if resource.id in overrides_by_pre_id:
+                resource.id = overrides_by_pre_id[resource.id].post_id
+
+        id_counts = Counter(resource.id for resource in resources)
+        duplicates = [key for key, count in id_counts.items() if count > 1]
```

Comment:
@emord suggested to use `most_common` here. Was just curious how that would work, so I think this would be
```
# fetch most common id
most_common_id_and_count = id_counts.most_common(1)
if most_common_id_and_count and most_common_id_and_count[0][1] > 1:
    raise ResourceOverrideError("Duplicate resource id found: {}".format(most_common_id_and_count[0][0])
```
but guess we want to get all duplicate ids. Unless there is something different that can be done with `Counter` here.

Reply:
Yeah, that's what the code would look like - I tried it out and didn't think it was any more readable.

Reply:
Also, FWIW, `most_common` sorts elements by their frequency, but it does also include elements which aren't duplicated.  `counts.most_common(3)` returns the 3 most common elements.  I don't think `Counter` has anything which returns only duplicates, so I like how Jenny's done here, which is simplest.

Reply:
The small snippet i wrote was getting the most common element and if its occurrence is more than 1, that means it was a duplicate.
Agree with what has been done here is simple and cool